web/src/app/settings/items/hooks.ts

```diff
@@ -0,0 +1,191 @@
+  categoryId: number,
+  createdDate: string,
+  action?: string
+}
+
+type CreateItem = Pick<IItem, 'nsid' | 'category'>
+
+export const useHooks = () => {
+  const { control, handleSubmit, setValue, reset } = useForm();
+  const [isOpenModal, setIsOpenModal] = useState(false);
+  const [isEditMode, setIsEditMode] = useState(false);
+  const [selectedEditId, setSelectedEditId] = useState<number | undefined>(undefined)
+
+  const columnSchema: IColumnSchema<IItem>[] = [
+    {
+      key: "nsid",
+      label: "NSID"
+    },
+    {
+      key: "itemCode",
+      label: "Item Code"
+    },
+    {
+      key: "item",
+      label: "Item"
+    },
+    {
+      key: "category",
+      label: "Category"
+    },
+    {
+      key: "createdDate",
+      label: "Created Date"
+    },
+    {
+      key: "action",
+      label: "Action",
+      actions: [
+        {
+          name: "Edit",
+          handleClick: (value) => handleEditItem(value)
+        },
+      ],
+    },
+  ]
+
+  const dataSource: IItem[] = [
+    {
+      id: 1,
+      category: "Nickko Category",
+      item: "Nicko 1",
+      itemCode: "Nicko-2023",
+      createdDate: "2024/26/21",
+      nsid: "122342-A",
+      categoryId: 4,
+    },
+    {
+      id: 2,
+      category: "Nickko Category",
+      item: "Nicko 2",
+      itemCode: "Nicko-2023",
+      createdDate: "2024/26/21",
+      nsid: "122342-B",
+      categoryId: 5,
+    },
+    {
+      id: 3,
+      category: "Nickko Category",
+      item: "Nicko 3",
+      itemCode: "Nicko-2023",
+      createdDate: "2024/26/21",
+      nsid: "122342-C",
+      categoryId: 6,
+    },
+  ]
+
+  const options: IOptionsProps[] = [
+    {
+      key: 4,
+      value: "Office Equipment  supplies"
+    },
+    {
+      key: 5,
+      value: "Motorpool equipment  Supplies"
+    },
+    {
+      key: 6,
+      value: "Vehicle repair  maintenance"
+    },
+    {
+      key: 7,
+      value: "Ricemill Equipment & Supplies"
+    },
+    {
+      key: 8,
+      value: "Utility Materials & Supplies"
+    },
+    {
+      key: 9,
+      value: "Various Items"
+    },
+  ]
+
+  const fields: IField<ICustomInputProps | ICustomSelectProps>[] = [
+    {
+      fieldType: "text",
+      fieldProps: {
+        label: "NSID",
+        name: "nsid",
+        control: control,
+        rules: { required: true }
+      },
+    },
+    {
+      fieldType: "dropdown",
+      fieldProps: {
+        label: "Category",
+        name: "category",
+        control: control,
+        options,
+        rules: { required: true }
+      }
+    },
+    {
+      fieldType: "text",
+      fieldProps: {
+        label: "Item",
+        name: "item",
+        control: control,
+        rules: { required: true }
+      },
+    },
+  ];
+
+  const handleCreateItem = () => {
+    setIsOpenModal(true)
+  }
+
+  const handleEditItem = (item: IItem) => {
+    setIsOpenModal(true)
+    setIsEditMode(true)
+    setSelectedEditId(item.id);
+    setValue('item', item.item)
```

Comment:
use forof
e apil na dri and id no need nata mag butang og setter

for key of Object.key(item) {
   setValue(key, item[key as keyof typeof item])
Comment

}
```suggestion
    setValue('item', item.item)
```